### PR TITLE
Add Slack reporter

### LINKED ISSUE #1643: Report test failures to slack
We current send test failures to google group istio-oncall, but I don't think many look at this. I don't think testgrid supports slack, but we can set up the slack Email app to get an email that will forward to slack, where it will get noticed

We have many daily releases and no one notices

### PR DESCRIPTION
**Pull request by scottilee**
Per https://github.com/kubernetes/test-infra/blob/master/prow/cmd/crier/README.md#slack-reporter this reports test failures and errors to Slack.

Fixes https://github.com/istio/test-infra/issues/1643.

/assign @howardjohn

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -55,6 +55,19 @@ prowjob_namespace: default
 pod_namespace: test-pods
 log_level: debug
 
+slack_reporter:
+  job_types_to_report:
+  - presubmit
```
Comment (geeknoid) [inline prow/config.yaml]: I don't think we want to report presubmit failures. That'd be a lot of noise.

Comment (fejta) [reply prow/config.yaml]: presubmits might be okay if it messaged the person who authored the PR, but that isn't how it works.

Comment (scottilee) [reply prow/config.yaml]: Removed.

**Review by howardjohn — APPROVED** [review]

**Review by clarketm — APPROVED** [review]

**Pull request merged**